server/src/controllers/ChapterUser/resolver.ts

```diff
@@ -54,7 +54,6 @@ export class ChapterUserResolver {
     });
   }
 
-  @Authorized(Permission.ChapterJoin)
```

Comment:
I have removed it because as long as you are logged in, there shouldn't be a blocked for you joining and leaving a chapter, as pointed out by Oliver.

That being said, I am wondering about RSVPs, because people need to be Auth for chapter to book events for them. Do we need to add a blocker for RSVP, checking if they auth us to update their Google calendar?

Reply:
Thing is, all members have the `JoinChapter` permission.  With the exception of banned users (they're restricted and maybe we should kick them out of the chapter, too, I don't think that's been settled yet) there are no restrictions imposed by this decorator.

> That being said, I am wondering about RSVPs, because people need to be Auth for chapter to book events for them.

If someone leaves a chapter, they're leaving the events, too.  This means their RSVPs will cease to exist and the Google Calendar Event needs updating to reflect the fact they're not longer attending.

> Do we need to add a blocker for RSVP, checking if they auth chapter to update their Google calendar?

No, because it's not their calendar.  They're invited to an event on the Chapter's Calendar (technically the calendar of the owner that first authenticated with Google), so they don't have to authorize anything.  It's the same thing as if I invited you to an event on my calendar - you don't have to authorize that, you just get an invite in the mail.

Reply:
I have no idea how to update google calendar 😅

Reply:
does deleting google calendar need to be added to removing RSVP from Event, as well?

Reply:
The resolver has a helper `updateCalendarEventAttendees` which handles this.  Once the _event_users_ have been updated, this can be called to update the Google calendar with those changes.

Have a look at the logic for `deleteRsvp`.  It's quite similar to `leaveChapter` at least as far as calendar event attendees.

Reply:
Copying deleteRsvp lead to this, which didn't create issue. but here is an issue with the logic because it delete one calendar instead of all of them

```ts
await prisma.event_users.delete({
      where: { user_id_event_id: { user_id: ctx.user.id, event_id: { chapter_id: chapterId } } },
      select: {
        event: {
          select: {
            calendar_event_id: true,
            chapter: {
              select: { calendar_id: true },
            },
          },
        },
      },
    });
```

I tried to include, calendar_id and a like this. but getting type error

```ts
     await prisma.event_users.deleteMany({
       where: {
         user_id: ctx.user.id,
         event: { chapter_id: chapterId },
         calendar_event_id: { chapter_id: chapterId },
         chapter: {
          select: { calendar_id: true },
        } 
       },       
     });
```

here is the type error.

>   Object literal may only specify known properties, and 'calendar_event_id' does not exist in type 'event_usersWhereInput'

Wondering how to include the `calendar_event_id` without adding field to `event_user`?

Reply:
What is `calendar_event_id` expected to do here?

Reply:
find events in the calendars which has a relation with chapterId

Reply:
`calendar_event_id` doesn't have relation with chapter